libraries/standard/serializer/src/cbor/iot_serializer_tinycbor_decoder.c

```diff
@@ -296,9 +315,28 @@ static IotSerializerError_t _createDecoderObject( _cborValueWrapper_t * pCborVal
                          * If its a finite length text/byte string, and user have passed a null length buffer,
                          * we avoid copying the string by storing pointer to the start of the string.
                          */
-                        pDecoderObject->u.value.u.string.pString =
-                            ( uint8_t * ) ( cbor_value_get_next_byte( &next ) -
-                                            ( pDecoderObject->u.value.u.string.length ) );
+
+                        /* There is a special case for the last/ending binary/text string element
+                         * in an indefinite length container as indefinite length containers end with the
+                         * "break" byte (0xFF) to indicate the end of the container.
+                         * (Refer to the specification for "Indefinite-Length Arrays and Maps"
+                         * https://tools.ietf.org/html/rfc7049#section-2.2.2 )
+                         * Thus, when calculating the starting string location for this special case,
+                         * we have to account for the byte. */
+                        uint8_t * pPossibleBreakByte = ( uint8_t * ) ( next.ptr - 1 );
```

Comment:
This is an issue we are working on fixing and this does not seem to be the right patch!

Reply:
This patch (to our wrapper of the tinycbor decoder) was merged last year (https://github.com/aws/aws-iot-device-sdk-embedded-C-staging/pull/7) to support parsing of indefinite length containers by the Provisioning library